src/unitxt/artifact.py

```diff
@@ -237,6 +244,11 @@ def from_dict(cls, d, overwrite_args=None):
     @classmethod
     def load(cls, path, artifact_identifier=None, overwrite_args=None):
         d = artifacts_json_cache(path)
+        if "artifact_linked_to" in d and d["artifact_linked_to"] is not None:
```

Comment:
I think  we should check if it is an instance of of ArtifactLink instead of checking for fields.